src/ts/components/smallcenteredplaybacktogglebutton.ts

```diff
@@ -0,0 +1,121 @@
+import {PlaybackToggleButton, PlaybackToggleButtonConfig} from './playbacktogglebutton';
+import {DOM} from '../dom';
+import {UIInstanceManager} from '../uimanager';
+import { PlayerAPI, PlayerEventBase, WarningEvent } from 'bitmovin-player';
+import { i18n } from '../localization/i18n';
+
+export class SmallCenteredPlaybackToggleButton extends PlaybackToggleButton {
+
+  constructor(config: PlaybackToggleButtonConfig = {}) {
+    super(config);
+
+    this.config = this.mergeConfig(config, {
+      cssClass: 'ui-smallcenteredplaybacktogglebutton',
+      text: i18n.getLocalizer('playPause'),
+      role: 'button',
+    }, this.config);
+  }
+
+  configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
+    // Update button state through API events
+    super.configure(player, uimanager, false);
+
+    // Set enterFullscreenOnInitialPlayback if set in the uimanager config
+    if (typeof uimanager.getConfig().enterFullscreenOnInitialPlayback === 'boolean') {
+      this.config.enterFullscreenOnInitialPlayback = uimanager.getConfig().enterFullscreenOnInitialPlayback;
+    }
+
+    let togglePlayback = () => {
+      if (player.isPlaying() || this.isPlayInitiated) {
+        player.pause('ui');
+      } else {
+        player.play('ui');
+      }
+    };
+
+    let firstPlay = true;
+
+    this.onClick.subscribe(() => {
+      togglePlayback();
+
+      if (firstPlay && this.config.enterFullscreenOnInitialPlayback) {
+        player.setViewMode(player.exports.ViewMode.Fullscreen);
+      }
+    });
+
+    player.on(player.exports.PlayerEvent.Play, () => {
+      // Playback has really started, we can disable the flag to switch to normal toggle button handling
+      firstPlay = false;
+    });
+
+    player.on(player.exports.PlayerEvent.Warning, (event: WarningEvent) => {
+      if (event.code === player.exports.WarningCode.PLAYBACK_COULD_NOT_BE_STARTED) {
+        // if playback could not be started, reset the first play flag as we need the user interaction to start
+        firstPlay = true;
+      }
+    });
+
+    const suppressPlayButtonTransitionAnimation = () => {
+      // Disable the current animation
+      this.setTransitionAnimationsEnabled(false);
+
+      // Enable the transition animations for the next state change
+      this.onToggle.subscribeOnce(() => {
+        this.setTransitionAnimationsEnabled(true);
+      });
+    };
+
+    // Hide the play button animation when the UI is loaded (it should only be animated on state changes)
+    suppressPlayButtonTransitionAnimation();
+
+    const isAutoplayEnabled = player.getConfig().playback && Boolean(player.getConfig().playback.autoplay);
```

Comment:
```suggestion
    const isAutoplayEnabled = player.getConfig().playback && Boolean(player.getConfig().playback?.autoplay);
```
There is not necessarily a playback config.

Reply:
Already checking it with `player.getConfig().playback && ...` but i will change it to `const isAutoplayEnabled = Boolean(player.getConfig().playback?.autoplay);`